model/atmosphere/dycore/docs/icon4py_sphinx.py

```diff
@@ -0,0 +1,364 @@
+        if current_block:
+            comment_blocks.append('\n'.join(current_block))
+        return comment_blocks
+
+    def format_docstring_block(self, docstr_lines, keyword):
+        # Clean up and format
+        if docstr_lines[0].strip().startswith(f'# {keyword}:'):
+            docstr_lines.pop(0) # remove the {keyword} prefix
+        while docstr_lines[0].strip() == '#':
+            # strip empty lines from the beginning
+            docstr_lines.pop(0)
+        # strip leading and trailing whitespace of every line (maintain indentation)
+        # as well as comment character and space (indent+2)
+        indent = len(docstr_lines[0]) - len(docstr_lines[0].lstrip(' '))
+        docstr_lines = [line[min(indent+2,len(line)):].rstrip() for line in docstr_lines]
+        # strip empty lines from the end
+        while docstr_lines[-1] == '':
+            docstr_lines.pop(-1)
+        return docstr_lines
+    
+    def add_header(self, docstr_lines, method_info):
+        # Add a title with ReST formatting
+        method_name = method_info['module_local_name'] + '.' + method_info['method_name']
+        method_full_name = method_info['module_full_name'] + '.' + method_name
+        title = f':meth:`{method_name}()<{method_full_name}>`'
+        docstr_lines.insert(0, title)
+        docstr_lines.insert(1, '='*len(title))
+        docstr_lines.insert(2, '')
+        return docstr_lines
+
+    def add_footer(self, docstr_lines):
+        # Add a horizontal line at the end of the docstring
+        docstr_lines.append('')
+        docstr_lines.append('.. raw:: html')
+        docstr_lines.append('')
+        docstr_lines.append('   <hr>')
+        # and add one empty line at the end
+        docstr_lines.append('')
+        return docstr_lines
+
+    def process_scidocstrlines(self, docstr_lines, next_method_info):
+        # "Special" treatment of specific lines / blocks
+
+        latex_multiline_block = False
+        past_inputs = False
+
+        def insert_before_first_non_space(s, char_to_insert):
+            for i, char in enumerate(s):
+                if char != ' ':
+                    return s[:i] + char_to_insert + s[i:]
+            return s
+
+        for iline, line in enumerate(docstr_lines):
+
+            # Make collapsible Inputs section
+            if line.startswith('Inputs:'):
+                past_inputs = True
+                docstr_lines[iline] = '.. collapse:: Inputs'
+                docstr_lines.insert(iline+1, '')
+
+            # Identify LaTeX multiline blocks and align equations to the left
+            elif '$$' in line:
+                if not latex_multiline_block and '\\\\' in docstr_lines[iline+1]:
+                    latex_multiline_block=True
+                    continue
+                else:
+                    latex_multiline_block=False
+            if latex_multiline_block:
+                docstr_lines[iline] = insert_before_first_non_space(line, '&')
```

Comment:
If I properly understood the logic of this conditions I think it can be simplified to something like the following code, which adds the `&` char at the beginning of the lines inside a multiline math block

```suggestion
            elif '$$' in line:
                latex_math_block = not latex_math_block
            if latex_math_block and line.rstrip().endswith(r'\\'):  # multiline math block
                start_idx = len(line) - len(line.lstrip()) 
                docstr_lines[line_num] = f"{line[:start_idx]}&{line[start_idx:]}"
```

Also the `insert_before_first_non_space()` looks unnecessary to me.